services/QuillLMS/db/migrate/20230731184420_create_diagnostic_question_skills.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class CreateDiagnosticQuestionSkills < ActiveRecord::Migration[6.1]
+  def change
+    create_table :diagnostic_question_skills do |t|
+      t.string :name, null: false
+      t.references :question, index: true, foreign_key: true, null: false
+      t.references :skill_group, index: true, foreign_key: true, null: false
+
+      t.timestamps
+    end
```

Comment:
Given that, from what I can tell of the use-case, there will only ever be a single skill associated with any given question, what do you think about the idea of just storing a `skill_group_id` on the `questions` table itself?  It reduces flexibility in the case that we ever want to do something different, but it does let us avoid adding an additional model that we may not really need?  (That also lets us avoid an additional index and foreign key required to tie this record to `questions`, but given how small the `questions` table is, I doubt that's a huge deal.

Not a direct objection to this approach, but it felt like I should check.

Reply:
I think that's a really good idea, and will more directly reflect the frontend (where what used to be referred to as "skills" is now going to be referred to as "questions"). I'll go ahead and make that change now.

Reply:
Update after discussing with Hannah: since questions can be used in multiple diagnostics, we do need this join table so that the same question can be linked to multiple skill groups.